Example/Example/Shared/Networking.swift

```diff
@@ -9,100 +9,86 @@
-  url: URL(string: "http://localhost:3000")!,
-  resolvingAgainstBaseURL: true)!
-private let checkoutsPath = "/checkouts"
-private let configurationPath = "/configuration"
 
 private struct CheckoutsRequest: Encodable {
   let email: String
   let amount: String
 }
 
-private struct CheckoutsResponse: Decodable {
+struct CheckoutsResponse: Decodable {
   let url: URL
 }
 
 struct ConfigurationResponse: Decodable {
-
   let minimumAmount: Money?
   let maximumAmount: Money
 
   struct Money: Decodable {
     let amount: String
     let currency: String
   }
-
 }
 
 enum NetworkError: Error {
   case malformedUrl
   case unknown
 }
 
-func getCheckoutURL(
-  with email: String,
-  for amount: String,
-  completion: @escaping (Result<URL, Error>) -> Void
-) {
-  let baseUrl = URL(string: "http://\(Settings.host):\(Settings.port)")
-  var urlComponents = baseUrl.flatMap { URLComponents(url: $0, resolvingAgainstBaseURL: true) }
-  urlComponents?.path = checkoutsPath
-
-  guard let url = urlComponents?.url else {
-    completion(.failure(NetworkError.malformedUrl))
-    return
-  }
-
-  var request = URLRequest(url: url)
-  request.httpMethod = "POST"
-
-  // A failed encoding operation here would represent programmer error
-  // swiftlint:disable:next force_try
-  request.httpBody = try! JSONEncoder().encode(CheckoutsRequest(email: email, amount: amount))
-  request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+struct Networking {
+  typealias Completion = (Result<Data, Error>) -> Void
 
-  let task = session.dataTask(with: request) { data, _, error in
-    guard error == nil, let data = data else {
-      return completion(.failure(error ?? NetworkError.unknown))
-    }
+  var configuration: (_ completion: @escaping Completion) -> Void
+  var checkout: (_ email: String, _ amount: String, _ completion: @escaping Completion) -> Void
+}
 
-    do {
-      let response = try JSONDecoder().decode(CheckoutsResponse.self, from: data)
-      completion(.success(response.url))
-    } catch {
-      completion(.failure(error))
+extension Networking {
+  static let live = Self(
+    configuration: { completion in
+      do {
+        session.fire(try /"configuration", completion: completion)
+      } catch {
+        completion(.failure(error))
+      }
+    },
+    checkout: { email, amount, completion in
+      do {
+        var request = URLRequest(url: try /"checkouts")
+        request.httpMethod = "POST"
+        request.httpBody = try JSONEncoder().encode(CheckoutsRequest(email: email, amount: amount))
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        session.fire(request, completion: completion)
+      } catch {
+        completion(.failure(error))
+      }
     }
-  }
-
-  task.resume()
+  )
 }
 
-func getConfiguration(
-  completion: @escaping (Result<ConfigurationResponse, Error>) -> Void
-) {
+prefix operator /
+private prefix func / (path: String) throws -> URL {
```

Comment:
The operator doesn't look too bad in context. I understand it's making use of the semantics of `/` being used in URLs to try and convey that it returns a URL. I can help that feel a private function on the `APIClient` would be a better fit though? Or even a private `Endpoint` enum with a computed `url` property?